packages/social-metadata-previews/src/shared/ImageSelect.js

```diff
@@ -17,32 +37,46 @@ import {
  *
  * @returns {React.Component} A fragment with a title, optional warnings and an image selection button.
  */
-const ImageSelect = ( { title, warnings, onClick } ) =>
+const ImageSelect = ( { title, warnings, onClick, imageSelected, onRemoveButtonClick } ) =>
 	<Fragment>
 		<SimulatedLabel>
 			{ title }
 		</SimulatedLabel>
 		{
-			warnings.length > 0 &&
+			warnings.length > 0 && imageSelected &&
 			warnings.map( ( warning, index ) => <Alert key={ `warning${ index }` } type="warning">
 				{ warning }
 			</Alert> )
 		}
 		<StandardButton
 			onClick={ onClick }
 		>
-			Select image
+			{
+				imageSelected
+					? __( "Replace image", "yoast-components" )
+					: __( "Select image", "yoast-components" )
+			}
 		</StandardButton>
+		{
+			imageSelected && <UndoButton
```

Comment:
This could be my preference but i like this idea:


```suggestion
const renderUndoButton = ( onRemoveButtonClick ) => {
	return ( 
	<UndoButton
	onClick={ onRemoveButtonClick }
	>
	{ __( "Remove image", "yoast-components" ) }
	</UndoButton> )
};
	
imageSelected && renderUndoButton( onRemoveButtonClick )

```